### PR TITLE
Markdown improvements in nothing.md

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: docs/visual-basic/language-reference/nothing.md [hunk 1]
```diff
@@ -1,54 +1,56 @@
...
+
 ## See also
 
-- [Dim Statement](../../visual-basic/language-reference/statements/dim-statement.md)
-- [Object Lifetime: How Objects Are Created and Destroyed](../../visual-basic/programming-guide/language-features/objects-and-classes/object-lifetime-how-objects-are-created-and-destroyed.md)
-- [Lifetime in Visual Basic](../../visual-basic/programming-guide/language-features/declared-elements/lifetime.md)
-- [Is Operator](../../visual-basic/language-reference/operators/is-operator.md)
-- [IsNot Operator](../../visual-basic/language-reference/operators/isnot-operator.md)
-- [Nullable Value Types](../../visual-basic/programming-guide/language-features/data-types/nullable-value-types.md)
+- [Dim Statement](statements/dim-statement.md)
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
- [Dim Statement](./statements/dim-statement.md)
```

File: docs/visual-basic/language-reference/nothing.md [hunk 2]
```diff
@@ -1,54 +1,56 @@
...
+- [Object Lifetime: How Objects Are Created and Destroyed](../programming-guide/language-features/objects-and-classes/object-lifetime-how-objects-are-created-and-destroyed.md)
+- [Lifetime in Visual Basic](../programming-guide/language-features/declared-elements/lifetime.md)
+- [Is Operator](operators/is-operator.md)
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
- [Is Operator](./operators/is-operator.md)
```

File: docs/visual-basic/language-reference/nothing.md [hunk 3]
```diff
@@ -1,54 +1,56 @@
...
+- [IsNot Operator](operators/isnot-operator.md)
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
- [IsNot Operator](./operators/isnot-operator.md)
```

File: docs/visual-basic/language-reference/nothing.md [hunk 4]
```diff
@@ -1,54 +1,56 @@
...
+
+ For strings in Visual Basic, the empty string equals `Nothing`. Therefore, `"" = Nothing` is true.
+
+ The following example shows comparisons that use the `Is` and `IsNot` operators:
+
+ [!code-vb[VbVbalrKeywords#9](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/Class4.vb#9)]
+
+ If you declare a variable without using an `As` clause and set it to `Nothing`, the variable has a type of `Object`. An example of this is `Dim something = Nothing`. A compile-time error occurs in this case when `Option Strict` is on and `Option Infer` is off.
+
+ When you assign `Nothing` to an object variable, it no longer refers to any object instance. If the variable had previously referred to an instance, setting it to `Nothing` does not terminate the instance itself. The instance is terminated, and the memory and system resources associated with it are released, only after the garbage collector (GC) detects that there are no active references remaining.
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
When you assign `Nothing` to an object variable, it no longer refers to any object instance. If the variable had previously referred to an instance, setting it to `Nothing` does not terminate the instance itself. The instance is terminated, and the memory and system resources associated with it are released, only after the garbage collector (GC) detects that there are no active references remaining.
```

File: docs/visual-basic/language-reference/nothing.md [hunk 5]
```diff
@@ -1,54 +1,56 @@
 ---
 title: "Nothing (Visual Basic)"
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
title: "Nothing keyword - Visual Basic"
```

File: docs/visual-basic/language-reference/nothing.md [hunk 6]
```diff
@@ -1,54 +1,56 @@
...
+f1_keywords:
   - "Nothing"
   - "vb.Nothing"
-helpviewer_keywords: 
+helpviewer_keywords:
   - "Nothing keyword [Visual Basic]"
   - "Nothing keyword [Visual Basic], syntax"
 ms.assetid: 06176e2d-bbf7-4a37-afaa-a86ad21ee99f
 ---
 # Nothing (Visual Basic)
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
# Nothing keyword (Visual Basic)
```

File: docs/visual-basic/language-reference/nothing.md [hunk 7]
```diff
@@ -1,54 +1,56 @@
...
-  
- When you assign `Nothing` to an object variable, it no longer refers to any object instance. If the variable had previously referred to an instance, setting it to `Nothing` does not terminate the instance itself. The instance is terminated, and the memory and system resources associated with it are released, only after the garbage collector (GC) detects that there are no active references remaining.  
-  
- `Nothing` differs from the <xref:System.DBNull> object, which represents an uninitialized variant or a nonexistent database column.  
-  
+> For non-nullable value types, `Nothing` in Visual Basic differs from `null` in C#. In Visual Basic, if you set a variable of a non-nullable value type to `Nothing`, the variable is set to the default value for its declared type. In C#, if you assign a variable of a non-nullable value type to `null`, a compile-time error occurs.
+
+## Remarks
+
+ `Nothing` represents the default value of a data type. The default value depends on whether the variable is of a value type or of a reference type.
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
`Nothing` represents the default value of a data type. The default value depends on whether the variable is of a value type or of a reference type.
```

File: docs/visual-basic/language-reference/nothing.md [hunk 8]
```diff
@@ -1,54 +1,56 @@
...
+
+ A variable of a *value type* directly contains its value. Value types include all numeric data types, `Boolean`, `Char`, `Date`, all structures, and all enumerations. A variable of a *reference type* stores a reference to an instance of the object in memory. Reference types include classes, arrays, delegates, and strings. For more information, see [Value Types and Reference Types](../programming-guide/language-features/data-types/value-types-and-reference-types.md).
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
A variable of a *value type* directly contains its value. Value types include all numeric data types, `Boolean`, `Char`, `Date`, all structures, and all enumerations. A variable of a *reference type* stores a reference to an instance of the object in memory. Reference types include classes, arrays, delegates, and strings. For more information, see [Value Types and Reference Types](../programming-guide/language-features/data-types/value-types-and-reference-types.md).
```

File: docs/visual-basic/language-reference/nothing.md [hunk 9]
```diff
@@ -1,54 +1,56 @@
...
+
+ If a variable is of a value type, the behavior of `Nothing` depends on whether the variable is of a nullable data type. To represent a nullable value type, add a `?` modifier to the type name. Assigning `Nothing` to a nullable variable sets the value to `null`. For more information and examples, see [Nullable Value Types](../programming-guide/language-features/data-types/nullable-value-types.md).
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
If a variable is of a value type, the behavior of `Nothing` depends on whether the variable is of a nullable data type. To represent a nullable value type, add a `?` modifier to the type name. Assigning `Nothing` to a nullable variable sets the value to `null`. For more information and examples, see [Nullable Value Types](../programming-guide/language-features/data-types/nullable-value-types.md).
```

File: docs/visual-basic/language-reference/nothing.md [hunk 10]
```diff
@@ -1,54 +1,56 @@
...
+
+ If a variable is of a value type that is not nullable, assigning `Nothing` to it sets it to the default value for its declared type. If that type contains variable members, they are all set to their default values. The following example illustrates this for scalar types.
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
If a variable is of a value type that is not nullable, assigning `Nothing` to it sets it to the default value for its declared type. If that type contains variable members, they are all set to their default values. The following example illustrates this for scalar types.
```

File: docs/visual-basic/language-reference/nothing.md [hunk 11]
```diff
@@ -1,54 +1,56 @@
...
+
+ [!code-vb[VbVbalrKeywords#7](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/Class2.vb#7)]
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
[!code-vb[VbVbalrKeywords#7](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/Class2.vb#7)]
```

File: docs/visual-basic/language-reference/nothing.md [hunk 12]
```diff
@@ -1,54 +1,56 @@
...
+
+ If a variable is of a reference type, assigning `Nothing` to the variable sets it to a `null` reference of the variable's type. A variable that is set to a `null` reference is not associated with any object. The following example demonstrates this:
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
If a variable is of a reference type, assigning `Nothing` to the variable sets it to a `null` reference of the variable's type. A variable that is set to a `null` reference is not associated with any object. The following example demonstrates this:
```

File: docs/visual-basic/language-reference/nothing.md [hunk 13]
```diff
@@ -1,54 +1,56 @@
...
+
+ [!code-vb[VbVbalrKeywords#8](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/class3.vb#8)]
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
[!code-vb[VbVbalrKeywords#8](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/class3.vb#8)]
```

File: docs/visual-basic/language-reference/nothing.md [hunk 14]
```diff
@@ -1,54 +1,56 @@
...
+
+ When checking whether a reference (or nullable value type) variable is `null`, do not use `= Nothing` or `<> Nothing`. Always use `Is Nothing` or `IsNot Nothing`.
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
When checking whether a reference (or nullable value type) variable is `null`, do not use `= Nothing` or `<> Nothing`. Always use `Is Nothing` or `IsNot Nothing`.
```

File: docs/visual-basic/language-reference/nothing.md [hunk 15]
```diff
@@ -1,54 +1,56 @@
...
+
+ For strings in Visual Basic, the empty string equals `Nothing`. Therefore, `"" = Nothing` is true.
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
For strings in Visual Basic, the empty string equals `Nothing`. Therefore, `"" = Nothing` is true.
```

File: docs/visual-basic/language-reference/nothing.md [hunk 16]
```diff
@@ -1,54 +1,56 @@
...
+
+ The following example shows comparisons that use the `Is` and `IsNot` operators:
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
The following example shows comparisons that use the `Is` and `IsNot` operators:
```

File: docs/visual-basic/language-reference/nothing.md [hunk 17]
```diff
@@ -1,54 +1,56 @@
...
+
+ [!code-vb[VbVbalrKeywords#9](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/Class4.vb#9)]
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
[!code-vb[VbVbalrKeywords#9](~/samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrKeywords/VB/Class4.vb#9)]
```

File: docs/visual-basic/language-reference/nothing.md [hunk 18]
```diff
@@ -1,54 +1,56 @@
...
+
+ If you declare a variable without using an `As` clause and set it to `Nothing`, the variable has a type of `Object`. An example of this is `Dim something = Nothing`. A compile-time error occurs in this case when `Option Strict` is on and `Option Infer` is off.
```
Comment (mairaw) [inline docs/visual-basic/language-reference/nothing.md]:
```suggestion
If you declare a variable without using an `As` clause and set it to `Nothing`, the variable has a type of `Object`. An example of this is `Dim something = Nothing`. A compile-time error occurs in this case when `Option Strict` is on and `Option Infer` is off.
```

**Review by mairaw — CHANGES_REQUESTED** [review]
Looks good @Youssef1313. Left a few suggestions!

**Comment by Youssef1313** [comment]
@mairaw, Should I start removing the space at the beginning of each paragraph ?

**Comment by Youssef1313** [comment]
Ping, @mairaw

**Review by mairaw — APPROVED** [review]
Looks good. Thanks @Youssef1313.

**Pull request merged**